skademlia/client.go

```diff
@@ -281,13 +286,19 @@ func (c *Client) connLoop(conn *grpc.ClientConn, id *ID) {
 			fallthrough
 		case connectivity.Shutdown:
 			c.peersLock.Lock()
-			delete(c.peers, conn.Target())
-			c.peersLock.Unlock()
+			if _, ok := c.peers[conn.Target()]; ok {
```

Comment:
Any reason why we shouldn't call onPeerLeave even if the peer isn't within the peers map?

Reply:
this is done to prevent more than one call to `onPeerLeave`

Reply:
Hm okay, would re-organize it a bit to make the happy path clear.

```go
c.peersLock.Lock()

_, existed := c.peers[conn.Target()]
delete(c.peers, conn.Target())
    
if existed && c.onPeerLeave != nil {
    c.onPeerLeave(conn, id)
}

c.peersLock.Unlock()
````

Reply:
initially i did this same way, but then i don't want to keep peers lock locked while `onPeerLeave` is running.